packages/launch/src/hooks/use-domain-search.ts

```diff
@@ -10,16 +10,19 @@ import { LAUNCH_STORE } from '../stores';
 import { useSite, useTitle } from './';
 import { isDefaultSiteTitle } from '../utils';
 
-export function useDomainSearch(): string {
+export function useDomainSearch(): { domainSearch: string; isLoading: boolean } {
 	const { domainSearch } = useSelect( ( select ) => select( LAUNCH_STORE ).getState() );
-	const { title } = useTitle();
-	const { currentDomainName } = useSite();
+	const { title, isLoading: isTitleLoading } = useTitle();
+	const { currentDomainName, isLoading: isSiteLoading } = useSite();
 
 	let search = domainSearch.trim() || title;
 
 	if ( ! search || isDefaultSiteTitle( { currentSiteTitle: search, exact: true } ) ) {
 		search = currentDomainName?.split( '.' )[ 0 ] ?? '';
 	}
 
-	return search;
+	return {
+		domainSearch: search,
+		isLoading: isTitleLoading || isSiteLoading,
```

Comment:
Both are always equal since they are reflecting the loading state of the site. Proposed a simplification in https://github.com/Automattic/wp-calypso/pull/47554/files#diff-fd3904ca53e493ed641e3deffd5efb5fd5a320c2aa489342a0c820f6dc573044R26